### PR TITLE
Added the extended api to the ros driver

### PR DESCRIPTION
**Pull request by ElizabethHollander**
Added the extended api to the ros driver, and uses that to get calibration values

Data skip can now be configured seperately for ir/rgb/depth

### CODE REVIEW

File: src/astra_driver.cpp [hunk 1]
```diff
@@ -278,7 +280,10 @@ void AstraDriver::configCb(Config &config, uint32_t level)
 
   use_device_time_ = config.use_device_time;
 
-  data_skip_ = config.data_skip+1;
+  // Todo: add seperate data skip for all streaming modes
+  ir_data_skip_ = config.ir_data_skip+1;
```
Comment (danielhailuA) [inline src/astra_driver.cpp:283]: Is TODO done here?

File: src/astra_driver.cpp [hunk 2]
```diff
@@ -571,41 +576,60 @@ void AstraDriver::newDepthFrameCallback(sensor_msgs::ImagePtr image)
...
-  // Aspect ratio for the camera center on Astra (and other devices?) is 4/3
-  // This formula keeps the principal point the same in VGA and SXGA modes
-  info->K[5] = (width * (3./8.)) - 0.5;
   info->K[8] = 1.0;
 
-  // No separate rectified image plane, so R = I
-  info->R.assign(0.0);
-  info->R[0] = info->R[4] = info->R[8] = 1.0;
+  if (is_ir_camera) {
+    info->D.assign(camera_params.l_k, camera_params.l_k + 5);
```
Comment (danielhailuA) [inline src/astra_driver.cpp:595]: The depth camera uses the IR camera distortion and K matrix values.

File: src/astra_driver.cpp (hunk 1 above)
Comment (ElizabethHollander) [reply src/astra_driver.cpp:283]: That's already done. Good catch, I'll remove the comment.

File: src/astra_driver.cpp (hunk 2 above)
Comment (ElizabethHollander) [reply src/astra_driver.cpp:595]: correct

**Review by danielhailuA — APPROVED** [review]

**Review by afakihcpr — APPROVED** [review]

**Pull request merged**